api/src/main/java/com/velocitypowered/api/event/player/PlayerChooseInitialServerEvent.java

```diff
@@ -54,6 +57,20 @@ public void setInitialServer(@Nullable RegisteredServer server) {
     this.initialServer = server;
   }
 
+  public Optional<Component> getReason() {
+    return Optional.ofNullable(reason);
+  }
+
+  /**
+   * Sets a custom disconnect reason for the player.
+   * Passing {@code null} will show the default reason.
+   *
+   * @param reason the disconnect reason to show to the player
+   */
```

Comment:
It would be good to detail the functionality of the method, something like: “if a valid server is not provided as the initial server, this message will be used to disconnect the player; if it is not present or null is used, the default message will be used”